### PR TITLE
feat(oas): declare x-codegen on Admin routes - PRO to R

### PR DESCRIPTION
**Pull request by patrick-medusajs**
### What

Declare `x-codegen` in OAS for Admin routes - PRO to R.

### Why

We are introducing a new `x-codegen` OpenApi extension, also known as _vendor extension_, in order to help with passing information down to code generators.

In our case, we wish to declare the `method` name that we would expect to call on a client. This mimics our current JS client package. 
E.g. `medusaClient.product.list` where `product` is the tag of the route and `list` is the x-codegen.method value.

We are also defining the name of a potential typed object for query parameters. OAS 3.0 does not allow to bundle query parameters under a single definition but it is not uncommon to see API clients handle all query parameters as a single typed object, like our JS client package. With x-codegen.queryParams, a code generator can create a named and typed object to bundle all query parameters for a given route. 
E.g. `medusaClient.customer.retrieve(id: string, queryParams: AdminGetCustomerParams)` 

### How

Declare `x-codegen` as an object with fields `method` and `queryParams` on all paths.

Match method and queryParams values with equivalent ones from our current JS client package.

### Test

* Ran OAS validator.
* Ran docs build script.

Expect no visible changes to the documentation.

### CODE REVIEW

File: packages/medusa/src/api/routes/admin/products/update-option.ts [hunk 1]
```diff
@@ -19,6 +19,8 @@ import { EntityManager } from "typeorm"
  *     application/json:
  *       schema:
  *         $ref: "#/components/schemas/AdminPostProductsProductOptionsOption"
+ * x-codegen:
+ *   method: updateOption
```
Comment (patrick-medusajs) [inline packages/medusa/src/api/routes/admin/products/update-option.ts:23]: https://github.com/medusajs/medusa/blob/5883bad1b31f211ad87f735e583129995fef4f37/packages/medusa-js/src/resources/admin/products.ts#L133-L138

File: packages/medusa/src/api/routes/admin/products/update-product.ts [hunk 2]
```diff
@@ -40,6 +40,8 @@ import { validator } from "../../../../utils/validator"
  *     application/json:
  *       schema:
  *         $ref: "#/components/schemas/AdminPostProductsProductReq"
+ * x-codegen:
+ *   method: update
```
Comment (patrick-medusajs) [inline packages/medusa/src/api/routes/admin/products/update-product.ts:44]: https://github.com/medusajs/medusa/blob/5883bad1b31f211ad87f735e583129995fef4f37/packages/medusa-js/src/resources/admin/products.ts#L39-L43

File: packages/medusa/src/api/routes/admin/products/update-variant.ts [hunk 3]
```diff
@@ -34,6 +34,8 @@ import { validator } from "../../../../utils/validator"
  *     application/json:
  *       schema:
  *         $ref: "#/components/schemas/AdminPostProductsProductVariantsVariantReq"
+ * x-codegen:
+ *   method: updateVariant
```
Comment (patrick-medusajs) [inline packages/medusa/src/api/routes/admin/products/update-variant.ts:38]: https://github.com/medusajs/medusa/blob/5883bad1b31f211ad87f735e583129995fef4f37/packages/medusa-js/src/resources/admin/products.ts#L105-L110

File: packages/medusa/src/api/routes/admin/publishable-api-keys/add-channels-batch.ts [hunk 4]
```diff
@@ -19,6 +19,8 @@ import PublishableApiKeyService from "../../../../services/publishable-api-key"
  *     application/json:
  *       schema:
  *         $ref: "#/components/schemas/AdminPostPublishableApiKeySalesChannelsBatchReq"
+ * x-codegen:
+ *   method: addSalesChannelsBatch
```
Comment (patrick-medusajs) [inline packages/medusa/src/api/routes/admin/publishable-api-keys/add-channels-batch.ts:23]: https://github.com/medusajs/medusa/blob/5883bad1b31f211ad87f735e583129995fef4f37/packages/medusa-js/src/resources/admin/publishable-api-keys.ts#L82-L86

File: packages/medusa/src/api/routes/admin/publishable-api-keys/create-publishable-api-key.ts [hunk 5]
```diff
@@ -15,6 +15,8 @@ import PublishableApiKeyService from "../../../../services/publishable-api-key"
  *       schema:
  *         $ref: "#/components/schemas/AdminPostPublishableApiKeysReq"
  * x-authenticated: true
+ * x-codegen:
+ *   method: create
```
Comment (patrick-medusajs) [inline packages/medusa/src/api/routes/admin/publishable-api-keys/create-publishable-api-key.ts:19]: https://github.com/medusajs/medusa/blob/5883bad1b31f211ad87f735e583129995fef4f37/packages/medusa-js/src/resources/admin/publishable-api-keys.ts#L49-L52

File: packages/medusa/src/api/routes/admin/publishable-api-keys/delete-channels-batch.ts [hunk 6]
```diff
@@ -19,6 +19,8 @@ import PublishableApiKeyService from "../../../../services/publishable-api-key"
  *     application/json:
  *       schema:
  *         $ref: "#/components/schemas/AdminDeletePublishableApiKeySalesChannelsBatchReq"
+ * x-codegen:
+ *   method: deleteSalesChannelsBatch
```
Comment (patrick-medusajs) [inline packages/medusa/src/api/routes/admin/publishable-api-keys/delete-channels-batch.ts:23]: https://github.com/medusajs/medusa/blob/5883bad1b31f211ad87f735e583129995fef4f37/packages/medusa-js/src/resources/admin/publishable-api-keys.ts#L91-L95

File: packages/medusa/src/api/routes/admin/publishable-api-keys/delete-publishable-api-key.ts [hunk 7]
```diff
@@ -10,6 +10,8 @@ import PublishableApiKeyService from "../../../../services/publishable-api-key"
  * x-authenticated: true
  * parameters:
  *   - (path) id=* {string} The ID of the PublishableApiKeys to delete.
+ * x-codegen:
+ *   method: delete
```
Comment (patrick-medusajs) [inline packages/medusa/src/api/routes/admin/publishable-api-keys/delete-publishable-api-key.ts:14]: https://github.com/medusajs/medusa/blob/5883bad1b31f211ad87f735e583129995fef4f37/packages/medusa-js/src/resources/admin/publishable-api-keys.ts#L66-L69

File: packages/medusa/src/api/routes/admin/publishable-api-keys/get-publishable-api-key.ts [hunk 8]
```diff
@@ -10,6 +10,8 @@ import PublishableApiKeyService from "../../../../services/publishable-api-key"
  * parameters:
  *   - (path) id=* {string} The ID of the PublishableApiKey.
  * x-authenticated: true
+ * x-codegen:
+ *   method: retrieve
```
Comment (patrick-medusajs) [inline packages/medusa/src/api/routes/admin/publishable-api-keys/get-publishable-api-key.ts:14]: https://github.com/medusajs/medusa/blob/5883bad1b31f211ad87f735e583129995fef4f37/packages/medusa-js/src/resources/admin/publishable-api-keys.ts#L20-L24

File: packages/medusa/src/api/routes/admin/publishable-api-keys/list-publishable-api-key-sales-channels.ts [hunk 9]
```diff
@@ -13,6 +13,9 @@ import { extendedFindParamsMixin } from "../../../../types/common"
  * parameters:
  *   - (path) id=* {string} The ID of the Publishable Api Key.
  *   - (query) q {string} Query used for searching sales channels' names and descriptions.
+ * x-codegen:
+ *   method: listSalesChannels
+ *   queryParams: GetPublishableApiKeySalesChannelsParams
```
Comment (patrick-medusajs) [inline packages/medusa/src/api/routes/admin/publishable-api-keys/list-publishable-api-key-sales-channels.ts:18]: https://github.com/medusajs/medusa/blob/5883bad1b31f211ad87f735e583129995fef4f37/packages/medusa-js/src/resources/admin/publishable-api-keys.ts#L100-L104

File: packages/medusa/src/api/routes/admin/publishable-api-keys/list-publishable-api-keys.ts [hunk 10]
```diff
@@ -16,6 +16,9 @@ import PublishableApiKeyService from "../../../../services/publishable-api-key"
  *   - (query) offset=0 {number} The offset of items in response
  *   - (query) expand {string} Comma separated list of relations to include in the results.
  *   - (query) fields {string} Comma separated list of fields to include in the results.
+ * x-codegen:
+ *   method: list
+ *   queryParams: GetPublishableApiKeysParams
```
Comment (patrick-medusajs) [inline packages/medusa/src/api/routes/admin/publishable-api-keys/list-publishable-api-keys.ts:21]: https://github.com/medusajs/medusa/blob/5883bad1b31f211ad87f735e583129995fef4f37/packages/medusa-js/src/resources/admin/publishable-api-keys.ts#L35-L38

File: packages/medusa/src/api/routes/admin/publishable-api-keys/revoke-publishable-api-key.ts [hunk 11]
```diff
@@ -11,6 +11,8 @@ import PublishableApiKeyService from "../../../../services/publishable-api-key"
  * parameters:
  *   - (path) id=* {string} The ID of the PublishableApiKey.
  * x-authenticated: true
+ * x-codegen:
+ *   method: revoke
```
Comment (patrick-medusajs) [inline packages/medusa/src/api/routes/admin/publishable-api-keys/revoke-publishable-api-key.ts:15]: https://github.com/medusajs/medusa/blob/5883bad1b31f211ad87f735e583129995fef4f37/packages/medusa-js/src/resources/admin/publishable-api-keys.ts#L74-L77

File: packages/medusa/src/api/routes/admin/publishable-api-keys/update-publishable-api-key.ts [hunk 12]
```diff
@@ -17,6 +17,8 @@ import PublishableApiKeyService from "../../../../services/publishable-api-key"
  *     application/json:
  *       schema:
  *         $ref: "#/components/schemas/AdminPostPublishableApiKeysPublishableApiKeyReq"
+ * x-codegen:
+ *   method: update
```
Comment (patrick-medusajs) [inline packages/medusa/src/api/routes/admin/publishable-api-keys/update-publishable-api-key.ts:21]: https://github.com/medusajs/medusa/blob/5883bad1b31f211ad87f735e583129995fef4f37/packages/medusa-js/src/resources/admin/publishable-api-keys.ts#L57-L61

File: packages/medusa/src/api/routes/admin/regions/add-country.ts [hunk 13]
```diff
@@ -19,6 +19,8 @@ import { validator } from "../../../../utils/validator"
  *     application/json:
  *       schema:
  *         $ref: "#/components/schemas/AdminPostRegionsRegionCountriesReq"
+ * x-codegen:
+ *   method: addCountry
```
Comment (patrick-medusajs) [inline packages/medusa/src/api/routes/admin/regions/add-country.ts:23]: https://github.com/medusajs/medusa/blob/5883bad1b31f211ad87f735e583129995fef4f37/packages/medusa-js/src/resources/admin/regions.ts#L103-L107

File: packages/medusa/src/api/routes/admin/regions/add-fulfillment-provider.ts [hunk 14]
```diff
@@ -19,6 +19,8 @@ import { validator } from "../../../../utils/validator"
  *     application/json:
  *       schema:
  *         $ref: "#/components/schemas/AdminPostRegionsRegionFulfillmentProvidersReq"
+ * x-codegen:
+ *   method: addFulfillmentProvider
```
Comment (patrick-medusajs) [inline packages/medusa/src/api/routes/admin/regions/add-fulfillment-provider.ts:23]: https://github.com/medusajs/medusa/blob/5883bad1b31f211ad87f735e583129995fef4f37/packages/medusa-js/src/resources/admin/regions.ts#L135-L139

File: packages/medusa/src/api/routes/admin/regions/add-payment-provider.ts [hunk 15]
```diff
@@ -19,6 +19,8 @@ import { validator } from "../../../../utils/validator"
  *     application/json:
  *       schema:
  *         $ref: "#/components/schemas/AdminPostRegionsRegionPaymentProvidersReq"
+ * x-codegen:
+ *   method: addPaymentProvider
```
Comment (patrick-medusajs) [inline packages/medusa/src/api/routes/admin/regions/add-payment-provider.ts:23]: https://github.com/medusajs/medusa/blob/5883bad1b31f211ad87f735e583129995fef4f37/packages/medusa-js/src/resources/admin/regions.ts#L181-L185

File: packages/medusa/src/api/routes/admin/regions/create-region.ts [hunk 16]
```diff
@@ -25,6 +25,8 @@ import { validator } from "../../../../utils/validator"
  *     application/json:
  *       schema:
  *         $ref: "#/components/schemas/AdminPostRegionsReq"
+ * x-codegen:
+ *   method: create
```
Comment (patrick-medusajs) [inline packages/medusa/src/api/routes/admin/regions/create-region.ts:29]: https://github.com/medusajs/medusa/blob/5883bad1b31f211ad87f735e583129995fef4f37/packages/medusa-js/src/resources/admin/regions.ts#L24-L27

File: packages/medusa/src/api/routes/admin/regions/delete-region.ts [hunk 17]
```diff
@@ -9,6 +9,8 @@ import RegionService from "../../../../services/region"
  * x-authenticated: true
  * parameters:
  *   - (path) id=* {string} The ID of the Region.
+ * x-codegen:
+ *   method: delete
```
Comment (patrick-medusajs) [inline packages/medusa/src/api/routes/admin/regions/delete-region.ts:13]: https://github.com/medusajs/medusa/blob/5883bad1b31f211ad87f735e583129995fef4f37/packages/medusa-js/src/resources/admin/regions.ts#L54-L57

File: packages/medusa/src/api/routes/admin/regions/get-fulfillment-options.ts [hunk 18]
```diff
@@ -10,6 +10,8 @@ import RegionService from "../../../../services/region"
  * x-authenticated: true
  * parameters:
  *   - (path) id=* {string} The ID of the Region.
+ * x-codegen:
+ *   method: retrieveFulfillmentOptions
```
Comment (patrick-medusajs) [inline packages/medusa/src/api/routes/admin/regions/get-fulfillment-options.ts:14]: https://github.com/medusajs/medusa/blob/5883bad1b31f211ad87f735e583129995fef4f37/packages/medusa-js/src/resources/admin/regions.ts#L166-L169

File: packages/medusa/src/api/routes/admin/regions/get-region.ts [hunk 19]
```diff
@@ -10,6 +10,8 @@ import RegionService from "../../../../services/region"
  * x-authenticated: true
  * parameters:
  *   - (path) id=* {string} The ID of the Region.
+ * x-codegen:
+ *   method: retrieve
```
Comment (patrick-medusajs) [inline packages/medusa/src/api/routes/admin/regions/get-region.ts:14]: https://github.com/medusajs/medusa/blob/5883bad1b31f211ad87f735e583129995fef4f37/packages/medusa-js/src/resources/admin/regions.ts#L68-L71

File: packages/medusa/src/api/routes/admin/regions/list-regions.ts [hunk 20]
```diff
@@ -47,6 +47,9 @@ import { validator } from "../../../../utils/validator"
  *      type: object
  *    required: false
  *    description: Date comparison for when resulting region was deleted, i.e. less than, greater than etc.
+ * x-codegen:
+ *   method: list
+ *   queryParams: AdminGetRegionsParams
```
Comment (patrick-medusajs) [inline packages/medusa/src/api/routes/admin/regions/list-regions.ts:52]: https://github.com/medusajs/medusa/blob/5883bad1b31f211ad87f735e583129995fef4f37/packages/medusa-js/src/resources/admin/regions.ts#L82-L85

File: packages/medusa/src/api/routes/admin/regions/remove-country.ts [hunk 21]
```diff
@@ -20,6 +20,8 @@ import RegionService from "../../../../services/region"
  *       externalDocs:
  *         url: https://en.wikipedia.org/wiki/ISO_4217#Active_codes
  *         description: See a list of codes.
+ * x-codegen:
+ *   method: deleteCountry
```
Comment (patrick-medusajs) [inline packages/medusa/src/api/routes/admin/regions/remove-country.ts:24]: https://github.com/medusajs/medusa/blob/5883bad1b31f211ad87f735e583129995fef4f37/packages/medusa-js/src/resources/admin/regions.ts#L119-L123

File: packages/medusa/src/api/routes/admin/regions/remove-fulfillment-provider.ts [hunk 22]
```diff
@@ -12,6 +12,8 @@ import RegionService from "../../../../services/region"
  * parameters:
  *   - (path) id=* {string} The ID of the Region.
  *   - (path) provider_id=* {string} The ID of the Fulfillment Provider.
+ * x-codegen:
+ *   method: deleteFulfillmentProvider
```
Comment (patrick-medusajs) [inline packages/medusa/src/api/routes/admin/regions/remove-fulfillment-provider.ts:16]: https://github.com/medusajs/medusa/blob/5883bad1b31f211ad87f735e583129995fef4f37/packages/medusa-js/src/resources/admin/regions.ts#L151-L155

File: packages/medusa/src/api/routes/admin/regions/remove-payment-provider.ts [hunk 23]
```diff
@@ -12,6 +12,8 @@ import RegionService from "../../../../services/region"
  * parameters:
  *   - (path) id=* {string} The ID of the Region.
  *   - (path) provider_id=* {string} The ID of the Payment Provider.
+ * x-codegen:
+ *   method: deletePaymentProvider
```
Comment (patrick-medusajs) [inline packages/medusa/src/api/routes/admin/regions/remove-payment-provider.ts:16]: https://github.com/medusajs/medusa/blob/5883bad1b31f211ad87f735e583129995fef4f37/packages/medusa-js/src/resources/admin/regions.ts#L197-L201

File: packages/medusa/src/api/routes/admin/regions/update-region.ts [hunk 24]
```diff
@@ -27,6 +27,8 @@ import { validator } from "../../../../utils/validator"
  *     application/json:
  *       schema:
  *         $ref: "#/components/schemas/AdminPostRegionsRegionReq"
+ * x-codegen:
+ *   method: update
```
Comment (patrick-medusajs) [inline packages/medusa/src/api/routes/admin/regions/update-region.ts:31]: https://github.com/medusajs/medusa/blob/5883bad1b31f211ad87f735e583129995fef4f37/packages/medusa-js/src/resources/admin/regions.ts#L39-L43

File: packages/medusa/src/api/routes/admin/return-reasons/create-reason.ts [hunk 25]
```diff
@@ -19,6 +19,8 @@ import { EntityManager } from "typeorm"
  *     application/json:
  *       schema:
  *         $ref: "#/components/schemas/AdminPostReturnReasonsReq"
+ * x-codegen:
+ *   method: create
```
Comment (patrick-medusajs) [inline packages/medusa/src/api/routes/admin/return-reasons/create-reason.ts:23]: https://github.com/medusajs/medusa/blob/5883bad1b31f211ad87f735e583129995fef4f37/packages/medusa-js/src/resources/admin/return-reasons.ts#L18-L21

File: packages/medusa/src/api/routes/admin/return-reasons/delete-reason.ts [hunk 26]
```diff
@@ -9,6 +9,8 @@ import { ReturnReasonService } from "../../../../services"
  * x-authenticated: true
  * parameters:
  *   - (path) id=* {string} The ID of the return reason
+ * x-codegen:
+ *   method: delete
```
Comment (patrick-medusajs) [inline packages/medusa/src/api/routes/admin/return-reasons/delete-reason.ts:13]: https://github.com/medusajs/medusa/blob/5883bad1b31f211ad87f735e583129995fef4f37/packages/medusa-js/src/resources/admin/return-reasons.ts#L48-L51

File: packages/medusa/src/api/routes/admin/return-reasons/get-reason.ts [hunk 27]
```diff
@@ -13,6 +13,8 @@ import { ReturnReasonService } from "../../../../services"
  * x-authenticated: true
  * parameters:
  *   - (path) id=* {string} The ID of the Return Reason.
+ * x-codegen:
+ *   method: retrieve
```
Comment (patrick-medusajs) [inline packages/medusa/src/api/routes/admin/return-reasons/get-reason.ts:17]: https://github.com/medusajs/medusa/blob/5883bad1b31f211ad87f735e583129995fef4f37/packages/medusa-js/src/resources/admin/return-reasons.ts#L62-L65

File: packages/medusa/src/api/routes/admin/return-reasons/list-reasons.ts [hunk 28]
```diff
@@ -12,6 +12,8 @@ import { Selector } from "../../../../types/common"
  * summary: "List Return Reasons"
  * description: "Retrieves a list of Return Reasons."
  * x-authenticated: true
+ * x-codegen:
+ *   method: list
```
Comment (patrick-medusajs) [inline packages/medusa/src/api/routes/admin/return-reasons/list-reasons.ts:16]: https://github.com/medusajs/medusa/blob/5883bad1b31f211ad87f735e583129995fef4f37/packages/medusa-js/src/resources/admin/return-reasons.ts#L75-L77

File: packages/medusa/src/api/routes/admin/return-reasons/update-reason.ts [hunk 29]
```diff
@@ -21,6 +21,8 @@ import { EntityManager } from "typeorm"
  *     application/json:
  *       schema:
  *         $ref: "#/components/schemas/AdminPostReturnReasonsReasonReq"
+ * x-codegen:
+ *   method: update
```
Comment (patrick-medusajs) [inline packages/medusa/src/api/routes/admin/return-reasons/update-reason.ts:25]: https://github.com/medusajs/medusa/blob/5883bad1b31f211ad87f735e583129995fef4f37/packages/medusa-js/src/resources/admin/return-reasons.ts#L33-L37

File: packages/medusa/src/api/routes/admin/returns/cancel-return.ts [hunk 30]
```diff
@@ -12,6 +12,8 @@ import { EntityManager } from "typeorm"
  * description: "Registers a Return as canceled."
  * parameters:
  *   - (path) id=* {string} The ID of the Return.
+ * x-codegen:
+ *   method: cancel
```
Comment (patrick-medusajs) [inline packages/medusa/src/api/routes/admin/returns/cancel-return.ts:16]: https://github.com/medusajs/medusa/blob/5883bad1b31f211ad87f735e583129995fef4f37/packages/medusa-js/src/resources/admin/returns.ts#L19-L22

File: packages/medusa/src/api/routes/admin/returns/list-returns.ts [hunk 31]
```diff
@@ -14,6 +14,9 @@ import { Return } from "../../../../models"
  * parameters:
  *   - (query) limit=50 {number} The upper limit for the amount of responses returned.
  *   - (query) offset=0 {number} The offset of the list returned.
+ * x-codegen:
+ *   method: list
+ *   queryParams: AdminGetReturnsParams
```
Comment (patrick-medusajs) [inline packages/medusa/src/api/routes/admin/returns/list-returns.ts:19]: https://github.com/medusajs/medusa/blob/5883bad1b31f211ad87f735e583129995fef4f37/packages/medusa-js/src/resources/admin/returns.ts#L49-L52

File: packages/medusa/src/api/routes/admin/returns/receive-return.ts [hunk 32]
```diff
@@ -24,6 +24,8 @@ import { isDefined } from "medusa-core-utils"
  *     application/json:
  *       schema:
  *         $ref: "#/components/schemas/AdminPostReturnsReturnReceiveReq"
+ * x-codegen:
+ *   method: receive
```
Comment (patrick-medusajs) [inline packages/medusa/src/api/routes/admin/returns/receive-return.ts:28]: https://github.com/medusajs/medusa/blob/5883bad1b31f211ad87f735e583129995fef4f37/packages/medusa-js/src/resources/admin/returns.ts#L34-L38

**Review by patrick-medusajs — COMMENTED** [review]
self-review: highlighting noteworthy code changes

**Review by shahednasser — APPROVED** [review]
LGTM

**Pull request merged**